snooty/parser.py

```diff
@@ -701,7 +687,10 @@ def __init__(
         substitution_nodes: Dict[str, SerializableType] = {}
         for k, v in self.config.substitutions.items():
             node, substitution_diagnostics = parse_rst(self.parser, root, v)
-            substitution_nodes[k] = node.ast
+            ast = cast(Any, node.ast)
+            # Extract text nodes from <root> and <paragraph> parent nodes
+            paragraph = ast["children"][0]
```

Comment:
What happens if an empty substitution is defined?

Reply:
Added handling!